plinth/modules/diaspora/__init__.py

```diff
@@ -0,0 +1,169 @@
+
+domain_name_file = "/etc/diaspora/domain_name"
+lazy_domain_name = None  # To avoid repeatedly reading from file
+
+
+def is_setup():
+    return os.path.exists(domain_name_file)
+
+
+def get_configured_domain_name():
+    if lazy_domain_name:
+        return lazy_domain_name
+
+    if not is_setup():
+        return ""
+
+    with open(domain_name_file) as dnf:
+        global lazy_domain_name
+        lazy_domain_name =  dnf.read().rstrip()
+        return lazy_domain_name
+
+
+version = 1
+
+title_en = 'Federated Social Network (diaspora*)'
+
+title = _(title_en)
+
+depends = ['apps']
+
+service = None
+
+managed_services = ['diaspora']
+
+managed_packages = ['diaspora']
+
+description = [
+    _('diaspora* is a decentralized social network where you can store and control your own data.'
+      ),
+    format_lazy(
+        'When enabled, the diaspora* pod will be available from '
+        '<a href="https://diaspora.{host}">diaspora.{host}</a> path on the web server.'.
+        format(host=get_configured_domain_name()))
+]
+
+
+def init():
+    """Initialize the Diaspora module."""
+    menu = cfg.main_menu.get('apps:index')
+    menu.add_urlname(title, 'glyphicon-thumbs-up', 'diaspora:index')
+
+    global service
+    setup_helper = globals()['setup_helper']
+    if setup_helper.get_state() != 'needs-setup':
+        service = service_module.Service(
+            managed_services[0],
+            title,
+            ports=['http', 'https'],
+            is_external=True,
+            is_enabled=is_enabled,
+            enable=enable,
+            disable=disable)
+
+        if is_enabled():
+            add_shortcut()
+
+
+def setup(helper, old_version=None):
+    """Install and configure the module."""
+    helper.call('pre', actions.superuser_run, 'diaspora', ['pre-install'])
+    helper.install(managed_packages)
+    helper.call('custom_config', actions.superuser_run, 'diaspora',
+                ['disable-ssl'])
+    helper.call('post', actions.superuser_run, 'diaspora', ['enable'])
+    global service
+    if service is None:
+        service = service_module.Service(
+            managed_services[0],
+            title,
+            ports=['http', 'https'],
+            is_external=True,
+            is_enabled=is_enabled,
+            enable=enable,
+            disable=disable)
+    helper.call('post', service.notify_enabled, None, True)
+    helper.call('post', add_shortcut)
+
+
+def get_domain_names():
+    """Return the domain name(s)"""
+    results = []
+
+    for domain_type, domains in names.domains.items():
+        if domain_type == 'hiddenservice':
+            continue
+        for domain in domains:
+            results.append((domain, domain))
+
+    return results
+
+
+def add_shortcut():
+    """Add shortcut to diaspora on the Plinth homepage"""
+    frontpage.add_shortcut(
+        'diaspora',
+        title,
+        url='https://diaspora.{}'.format(get_configured_domain_name()),
```

Comment:
This will also be a bad url, if no domain is set. Perhaps we can change when the shortcut is added to the frontpage -- only add after a domain is properly configured.